backend/indexer/internal/processor/handlers/fault_record_handler.go

```diff
@@ -150,6 +150,8 @@ func (h *FaultRecordHandler) HandleEvent(ctx context.Context, eventLog *types.Lo
 
 	challengeEpoch := proofSet.NextChallengeEpoch
 	proofSetOwner := proofSet.Owner
+	totalDataSize := proofSet.TotalDataSize
```

Comment:
One tricky thing is that challenge generation uses a version of total data size that is only updated on every call to nextProvingPeriod, not on the call to AddRoots.  I'm not sure how TotalDataSize is updated but if it is every AddRoot then we'll need to modify this / keep a local "challengeRange" variable to generate the correct challenges.

Reply:
Looking into this more we indeed update this variable every add which departs from challengeRange values.  I think the best way to do this is 
1) add ChallengeRange to the table init to 0
2) every nextProvingPeriod message set challengeRange = totalDataSize